src/layout/IHyprLayout.hpp

```diff
@@ -211,6 +212,11 @@ class IHyprLayout {
     */
     virtual bool updateDragWindow();
 
+    /*
+        Triggers a window snap event
+    */
+    static void performSnap(Vector2D& sourcePos, Vector2D& sourceSize, PHLWINDOW DRAGGINGWINDOW, const eMouseBindMode MODE, const int CORNER, const Vector2D& BEGINSIZE);
```

Comment:
I'd say make it a virtual. Maybe some layouts will want to override this, who knows?